File: trab_phyton/final_phyton/com.example.final_project_fxml/GuestListController.py
import pickle
from hostelapp.model import Guest, Hostel
from tkinter import Listbox

class GuestListController:
    def __init__(self):
        self.guest_list = Listbox()  

        self.FILE_PATH = "guests.dat"

        self.loadGuestsFromFile()

    def loadGuestsFromFile(self):
        try:
            with open(self.FILE_PATH, "rb") as file:
                guests = pickle.load(file)
                self.guest_list.delete(0, 'end')
                for guest in guests:
                    self.guest_list.insert('end', guest)
        except (IOError, pickle.UnpicklingError) as e:
            print("Erro ao carregar hóspedes do arquivo:", e)

    def saveGuestsToFile(self):
        hostel = Hostel.getHostel()
        guests = hostel.getAllGuests()

        try:
            with open(self.FILE_PATH, "wb") as file:
                pickle.dump(guests, file)
        except IOError as e:
            print("Erro ao salvar hóspedes no arquivo:", e)

    def listarTodosHospedes(self):
        hostel = Hostel.getHostel()
        lista_hospedes = hostel.getAllGuests()
        self.guest_list.delete(0, 'end')
        for guest in lista_hospedes:
            self.guest_list.insert('end', guest)
        self.saveGuestsToFile()

File: trab_java/final_project/src/main/java/com/example/final_project/GuestListController.java
package com.example.final_project;

import hostelapp.model.Guest;
import hostelapp.model.Hostel;
import javafx.collections.FXCollections;
import javafx.collections.ObservableList;
import javafx.fxml.FXML;
import javafx.scene.control.ListView;
import java.io.*;
import java.util.List;

public class GuestListController {

    @FXML
    private ListView<Guest> guest_list;

    private final String FILE_PATH = "guests.dat";

    public void loadGuestsFromFile() {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(FILE_PATH))) {
            List<Guest> guests = (List<Guest>) ois.readObject();
            ObservableList<Guest> observableList = FXCollections.observableArrayList(guests);
            guest_list.setItems(observableList);
        } catch (IOException | ClassNotFoundException e) {
            System.err.println("Error while loading guests from file: " + e.getMessage());
        }
    }

    public void saveGuestsToFile() {
        Hostel hostel = Hostel.getHostel();
        List<Guest> guests = hostel.getAllGuests();

        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(FILE_PATH))) {
            oos.writeObject(guests);
        } catch (IOException e) {
            System.err.println("Error while saving guests to file: " + e.getMessage());
        }
    }


    @FXML public void initialize(){
        loadGuestsFromFile();
    }

    public void listAllGuest(){
        Hostel hostel = Hostel.getHostel();
        List<Guest> lista = hostel.getAllGuests();
        ObservableList<Guest> observableList = FXCollections.observableArrayList(lista);
        guest_list.setItems(observableList);
        saveGuestsToFile();
    }



}